realsense_camera/src/base_nodelet.cpp

```diff
@@ -275,34 +276,67 @@ namespace realsense_camera
   {
     // print list of detected cameras
     std::vector<int> camera_type_index;
-    std::string detected_camera_msg = " - Detected the following cameras:";
+
+    // map camera names to validated firmware
+    std::map<std::string, std::string> camera_name_to_validated_firmware =
```

Comment:
Can this map definition be const and moved into constants.h? We've tried to avoid putting any hard-coded constants in the cpp files. Also, we've tried to avoid having any camera type specific code in the base nodelet, to keep it generic to all camera types.